### PR TITLE
feature/AK-34: Added AWS EC2 Open Custom Ports plugin, text cases

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

**Review by giorod3 — APPROVED** [review]

File: plugins/aws/ec2/openCustomPorts.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    category: 'EC2',
+    description: 'Ensure that defined custom ports are not open to public.',
+    more_info: 'Security groups should restrict access to ports from known networks.',
+    link: 'https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html',
+    recommended_action: 'Modify the security group to ensure the defined custom ports are not exposed publicly',
+    apis: ['EC2:describeSecurityGroups'],
+    settings: {
+        restricted_open_ports: {
+            name: 'Restricted Open Ports',
+            description: 'Comma separated list of ports that should be restricted to be publicly open',
```
Comment (matthewdfuller) [inline plugins/aws/ec2/openCustomPorts.js:15]:
```suggestion
            description: 'Comma separated list of ports that should be restricted and not publicly open',
```

File: plugins/aws/ec2/openCustomPorts.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            regex: '[a-zA-Z0-9,:]',
+            default: 'tcp:22,tcp:23'
```
Comment (matthewdfuller) [inline plugins/aws/ec2/openCustomPorts.js:17]: Let's change to empty string

File: plugins/aws/ec2/openCustomPorts.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        },
+    },
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
```
Comment (matthewdfuller) [inline plugins/aws/ec2/openCustomPorts.js:25]: Exit if settings.restricted_open_ports is empty

**Review by matthewdfuller — CHANGES_REQUESTED** [review]
@giorod3 - see comments

**Review by giorod3 — APPROVED** [review]

File: plugins/aws/ec2/openCustomPorts.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    category: 'EC2',
+    description: 'Ensure that defined custom ports are not open to public.',
+    more_info: 'Security groups should restrict access to ports from known networks.',
+    link: 'https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html',
+    recommended_action: 'Modify the security group to ensure the defined custom ports are not exposed publicly',
+    apis: ['EC2:describeSecurityGroups'],
+    settings: {
+        restricted_open_ports: {
+            name: 'Restricted Open Ports',
+            description: 'Comma separated list of ports that should be restricted and not publicly open',
```
Comment (matthewdfuller) [inline plugins/aws/ec2/openCustomPorts.js:15]:
```suggestion
            description: 'Comma separated list of ports that should be restricted and not publicly open. Example: tcp:80,tcp:443',
```

**Pull request merged**